### ISSUE TITLE
Remove jsDump

### ISSUE THREAD

**Issue by trentmwillis** [jquery/qunit#1034]
This is a long way off, but I wanted to open an issue to make sure we remember to remove `QUnit.jsDump` in 3.0 (we forgot it in 2.0).

[Reference code](https://github.com/jquery/qunit/blob/e2b38da0be2ab4dc6fe548910fe5f71c55321f95/src/core.js#L180-L181).

**Comment by jzaefferer** [comment]
Why not remove it in a another 2.x release? At least with explicit error handling? It was deprecated a long time ago and its in the 2.x upgrade guide: http://qunitjs.com/upgrade-guide-2.x/#replace-qunit-jsdump-with-qunit-dump

**Comment by trentmwillis** [comment]
I suppose we could consider this a bugfix (since it was supposed to be removed in 2.0.0) and just release it in the next patch version.

**Comment by nithmanoj** [comment]
Hello, I will love to work on this issue, this will be my first contribution in qunit. I think it is just removing one line code, please guide me if I understood anything wrong -

- [ ] Remove line `QUnit.jsDump = dump` from core.js

**Comment by platinumazure** [comment]
@nithmanoj Thanks for volunteering, we would love for you to contribute to QUnit! I just think we should get consensus on how best to handle this.

@jquery/qunit Okay folks, let's get consensus on this. Here are the open questions as I see them:

- [ ] Should we add an error message that can tell the user they should stop using `QUnit.jsDump`?
- [ ] Should we aim to put this in 3.0 or in a minor release (presumably the latter is only feasible if we add the error message)?

My vote is let's remove it, let's add an error message, and let's put it in a minor since folks already should have known it was going away. (Normally I'd say patch since it could be regarded as a bugfix, but adding an error message means enhancing functionality IMO, so I favor a minor release here.)

**Comment by jzaefferer** [comment]
👍  to replacing the alias with a function that throws an explicit error, as part of the next minor release.

**Comment by nithmanoj** [comment]
@platinumazure Thanks for such a great response, sir please review this checklist -

- [ ] Change `QUnit.jsDump` to function that throws error warning `Use QUnit.Dump instead of QUnit.jsDump`

please recommend your views for warning message to be thrown.

**Comment by trentmwillis** [comment]
Note `dump` and `jsDump` are Objects and not functions: https://github.com/jquery/qunit/blob/master/src/dump.js#L57

Thus in order to deprecate it will likely need to be a getter, instead of a function call. Such as we do here: https://github.com/jquery/qunit/blob/e2b38da0be2ab4dc6fe548910fe5f71c55321f95/src/export.js#L41-L46

**Comment by nithmanoj** [comment]
That was my wrong, @trentmwillis thanks for guiding me. Since `jsDump` is an object, what I guess `QUnit.reset` was a function,  do we need `Object.defineProperties` to error message for every property of `dump`.

**Comment by platinumazure** [comment]
@nithmanoj I think just putting a getter for `jsDump` should be sufficient, since we want to blacklist the entire object. @trentmwillis Sound okay?

```js
Object.defineProperty(QUnit, "jsDump", {
    get: function () { throw new Error(/* ... */); }
});
```

**Comment by platinumazure** [comment]
@trentmwillis Should we move this out of the v3.0 milestone if we've agreed to put this in as a semver-minor change?

**Comment by trentmwillis** [comment]
@platinumazure yep, removed it. I think the plan here is to release as a patch with the deprecation and then remove all together in 2.1, since the other errors are being removed then as well I believe. @leobalter can confirm/deny that.

**Comment by leobalter** [comment]
confirming 👍

**Comment by platinumazure** [comment]
Fixed via #1036!
